tests/functional/spec/components/address/address.spec.js

```diff
@@ -0,0 +1,58 @@
+import Address from "../../../generated_pages/address/address-block.page";
+import Summary from "../../../generated_pages/address/summary.page";
+
+describe("Address Answer Type", () => {
+  beforeEach("Launch survey", () => {
+    browser.openQuestionnaire("test_address.json");
+  });
+
+  describe("Given the user is on an address input question", () => {
+    it("When the user enters all address fields, Then, the summary displays the address fields", () => {
+      $(Address.Line1()).setValue("Evelyn Street");
+      $(Address.Line2()).setValue("Apt 7");
+      $(Address.Town()).setValue("Barry");
+      $(Address.Postcode()).setValue("CF63 4JG");
+
+      $(Address.submit()).click();
+      expect(browser.getUrl()).to.contain(Summary.pageName);
+      expect($(Summary.address()).getText()).to.contain("Evelyn Street\nApt 7\nBarry\nCF63 4JG");
+      expect($(Summary.address()).getHTML()).to.contain("Evelyn Street<br>Apt 7<br>Barry<br>CF63 4JG");
+    });
+  });
+
+  describe("Given the user is on an address input question", () => {
+    it("When the user enters only address line 1, Then, the summary only displays address line 1", () => {
+      $(Address.Line1()).setValue("Evelyn Street");
+
+      $(Address.submit()).click();
+      expect(browser.getUrl()).to.contain(Summary.pageName);
+      expect($(Summary.address()).getText()).to.contain("Evelyn Street");
+    });
+  });
+
+  describe("Given the user is on an address input question", () => {
+    it("When the user submits the page without entering any fields, Then, an error is displayed", () => {
```

Comment:
Any fields? shouldn't this be line 1, as that is what the code is validating against